cli/azd/pkg/apphost/generate.go

```diff
@@ -178,7 +178,19 @@ func ContainerAppManifestTemplateForProject(
 
 	var buf bytes.Buffer
 
-	tmplCtx := generator.containerAppTemplateContexts[projectName]
+	type yamlTemplateCtx struct {
+		genContainerAppManifestTemplateContext
+		TargetPortExpression string
+	}
+	tCtx := generator.containerAppTemplateContexts[projectName]
+	tmplCtx := yamlTemplateCtx{
+		genContainerAppManifestTemplateContext: tCtx,
+		TargetPortExpression:                   fmt.Sprintf("{{ targetPortOrDefault %d }}", tCtx.Ingress.TargetPort),
+	}
+	if tCtx.Ingress.TargetPort != 0 && !tCtx.Ingress.UsingDefaultPort {
```

Comment:
I'm wondering now if we still need to write `targetPortOrDefault <some non-negative number>` now that we're doing the work upfront to figure out whether the port in `containerapp.yaml` should be assigned a fixed value, or a dynamic one determined at deploy time.

In practice, isn't the value is either some literal like `3000`, or an expression like `.TargetPort` or `{{ targetPort }}` (which means figure out from container metadata, i.e. `dotnet publish`)?

Reply:
by using `targetPortOrDefault xxxx` , we are still adding a guard in case the port can't be resolved from dotnet publish (or any other build strategy). In those cases, we are setting a default in advance depending on the source type (project, container or dockerfile). Later, when `tarfetPortOrDefault` runs, we should not try to check if we are creating a containerApp for a project/container/dockerfile.

Reply:
How accurate would the "default port" we're suggesting be? Wearing a DevOps hat, a default that might not be accurate which will be applied to the running service is scarier than the deployment failing, so personally I would have preferred if the deployment failed outright.

Reply:
the default case is to expect for standard web apps listening on port 80 (or port 8080 for net projects).

The benefit here is directly for the simplest use case, like the aspire-starter, or publishing a popular container (like swagger-ui or something anything with web UI), so you don't need to manually set the target port.